src/simulators/tensor_network/matrix_product_state.cpp

```diff
@@ -92,6 +92,12 @@ cmatrix_t mul_matrix_by_lambda(const cmatrix_t &mat,
   if (lambda == rvector_t {1.0}) return mat;
   cmatrix_t res_mat(mat);
   uint_t num_rows = mat.GetRows(), num_cols = mat.GetColumns();
+
+  #ifdef _WIN32
+     #pragma omp parallel for
+  #else
+     #pragma omp parallel for collapse(2)
+  #endif
   for(uint_t row = 0; row < num_rows; row++) {
```

Comment:
We need to change the type of `row` to `int_t` due to incompatibilities with Windows OpenMP support.

Reply:
Corrected both places as you pointed out.
It is a problem to catch failures that only happen non-deterministically. I can only suggest running the unittest a few times.
We also use another kind of testing here - we create random circuits and then compare results between the mps and the statevector. If you are interested, I can add these somewhere as well.